File: 01. Leetcode/faangdsa/priyanka/arrays/03/rotateArray.java


public class rotateArray {
    public void rotate(int[] nums, int k) {
        int a = nums.length;
        k = k % a; 

        reverse(nums, 0, a-k-1);
        reverse(nums, a-k, a-1);
        reverse(nums, 0, a-1);
    }
    public void reverse(int[] nums, int i, int j){
        while(i<=j){
            swap(nums, i, j);
            i++;
            j--;
        }
    }
    public void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: 01. Leetcode/faangdsa/sachin/arrays/03/rotateArray.ts
function rotate(nums: number[], k: number): void {
    k %= nums.length;

    reverse(nums, 0, nums.length-1);
    reverse(nums, 0, k-1);
    reverse(nums, k, nums.length-1);
};

function reverse(nums: number[], start: number, end: number){
    while (start<end){
        let temp = nums[start];
        nums[start] = nums[end];
        nums[end] = temp;
        start++;
        end--;
    }
}